docs/articles/documentation/test-api/authentication/user-roles.md

```diff
@@ -10,7 +10,19 @@ to isolate authentication test actions and apply them easily whenever you need t
 
 A piece of logic that logs in a particular user is called a *role*. Define a role for each user account participating in your test.
 
-## Creating and Using Roles
+## Why Use Roles
+
+Roles are more than just another way to [extract reusable test logic](../../recipes/extract-reusable-test-code/README.md). They were specially designed for login operations and provide the following dedicated features:
+
+* **Object-based API.** Authentication data and logic are stored in an object that is easy to pass around and activate when needed.
+* **Single login.** Login actions are not repeated when you switch to a previously used role within the same session. If you activate a role in the [beforeEach](../test-code-structure.md#test-hooks) hook, login actions run once before the first test. The subsequent tests reuse authentication data so that it happens instantly.
+* **Automatic return.** The browser automatically navigates back to the page that was opened when you switched the role.
+* **Automatic logout.** No need to look for the *Log out* button.
```

Comment:
I've had doubts about this phrase since I read it first time. Do you mean using anonymous role?

Reply:
I mean that with when you use plain helpers for login actions, you would need to click *Log out* before you call the log in helper for a different account, wouldn't you?

With roles, you don't have to do this.

Am I wrong?

Reply:
Um, I didn't get it. You're right, but let add `...in case of switching between roles` or smth like that

Reply:
Well, Boris has a better point that we don't have automatic logout at all. I'll fix it.

Reply:
all right